website/pages/api-docs/secret/mongodbatlas/index.mdx

```diff
@@ -0,0 +1,249 @@
+
+### Sample Payload
+
+```json
+{
+  "project_id": "5cf5a45a9ccf6400e60981b6",
+  "roles": ["GROUP_CLUSTER_MANAGER"],
+  "cidr_blocks": ["192.168.1.3/32"],
+  "ip_addresses": ["192.168.1.3", "192.168.1.4"]
+}
+```
+
+```bash
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/mongodbatlas/roles/test-programmatic-key
+```
+
+### Sample Response
+```json
+{
+  "project_id": "5cf5a45a9ccf6400e60981b6",
+  "roles": ["GROUP_CLUSTER_MANAGER"],
+  "cidr_blocks": ["192.168.1.3/32"],
+  "ip_addresses": ["192.168.1.3", "192.168.1.4"],
+  "organization_id": "7cf5a45a9ccf6400e60981b7",
+  "ttl": "30m",
+  "max_ttl": "1h"
+}
+
+```
+
+## Read Programmatic API Key role
+
+| Method   | Path                         |
+| :--------------------------- | :--------------------- |
+| `Get`   | `/roles/:name`     |
+
+
+## Parameters
+
+`name` `(string <required>)` - Unique identifier name of the role name
+
+### Sample Payload
+
+```bash
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request GET \
+    --data @payload.json \
+    http://127.0.0.1:8200/mongodbatlas/roles/test-programmatic-key
+```
+
+### Sample Response
+```json
+{
+  "project_id": "5cf5a45a9ccf6400e60981b6",
+  "roles": ["GROUP_CLUSTER_MANAGER"],
+  "cidr_blocks": ["192.168.1.3/32"],
+  "ip_addresses": ["192.168.1.3", "192.168.1.4"],
+  "organization_id": "7cf5a45a9ccf6400e60981b7",
+  "ttl": "30m",
+  "max_ttl": "1h"
+}
+```
+
+## List Programmatic API Key role
+
+| Method   | Path                         |
+| :--------------------------- | :--------------------- |
+| `Get`   | `/roles`     |
+
+
+### Sample Payload
+
+```bash
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request GET \
+    --data @payload.json \
+    http://127.0.0.1:8200/mongodbatlas/roles
+```
+
+### Sample Response
+```json
+[
+  {
+    "project_id": "5cf5a45a9ccf6400e60981b6",
+    "roles": ["GROUP_CLUSTER_MANAGER"],
+    "cidr_blocks": ["192.168.1.3/32"],
+    "ip_addresses": ["192.168.1.3", "192.168.1.4"],
+    "organization_id": "7cf5a45a9ccf6400e60981b7",
+    "ttl": "30m",
+    "max_ttl": "1h"
+  },
+  {
+    "project_id": "5cf5a45a9ccf6400e60981b6",
+    "roles": ["READ"],
+    "cidr_blocks": ["192.168.1.3/35"],
+    "ip_addresses": ["192.168.1.5", "192.168.1.6"],
+    "organization_id": "7cf5a45a9ccf6400e60981b7",
+    "ttl": "30m",
+    "max_ttl": "1h"
+  }
+]
+
+```
+
+## Delete Programmatic API Key role
+
+| Method   | Path                         |
+| :--------------------------- | :--------------------- |
+| `DELETE`   | `/roles/:name`     |
+
+
+## Parameters
+
+`name` `(string <required>)` - Unique identifier name of the role name
+
+### Sample Payload
+
+```bash
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request DELETE \
+    --data @payload.json \
+    http://127.0.0.1:8200/mongodbatlas/roles/test-programmatic-key
+```
+
+### Sample Response
+```json
+{}
+```
```

Comment:
I have never tested with raw curl but I always assumed there was no body in a delete response. (I usually test with the CLI.) I can't find any other example responses in our docs for `DELETE` commands. Should we strip this? Looks a little odd to show an empty body. Or if we want to keep it, maybe we should note the returned status code.